codebase-pipelines/buildspec-manifests.yml

```
@@ -0,0 +1,28 @@
+version: 0.2
+
+env:
+  ${jsonencode({
+    exported-variables: flatten([ for env in environments : ["CLUSTER_NAME_${env}", [ for svc in services: "SERVICE_NAME_${env}_${svc}" ]]])
+  })}
```

Does this work? Might be a bit easier to follow. Not a deal breaker though.
```suggestion
  ${jsonencode({
    exported-variables: flatten([
      for env in environments : [
        "CLUSTER_NAME_${env}",
        [for svc in services: "SERVICE_NAME_${env}_${svc}"]
      ]
    ])
  })}
```